src/datamodel/builder/UMessage.cpp

```diff
@@ -34,7 +34,7 @@ UMessageBuilder UMessageBuilder::publish(v1::UUri&& topic) {
 
 UMessageBuilder UMessageBuilder::notification(v1::UUri&& source,
                                               v1::UUri&& sink) {
-	auto [srcOk, srcReason] = UriValidator::isValid(source);
+	auto [srcOk, srcReason] = UriValidator::isValidNotification(source);
```

Comment:
Looking at the [UAttributes spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uattributes.adoc#2-common-message-attributes), I think we should be checking both the source and sink against specific rules for each message type. However, we may also need additional UUri validators for this. I think I'll file a separate bug for this so it can be investigated more thoroughly.

Reply:
See #188

Reply:
In absence of additional UUri validators, what do you recommend we do in this PR ?